### PR TITLE
Add OpenAI-CLIP-Image-Text-Embeddings-ViT-Large-Patch14-336 Embeddings model

### PR DESCRIPTION
**Pull request by bhimar**
Registry Model: [model](https://ml.azure.com/registries/automl-images-dev/models/OpenAI-CLIP-Image-Text-Embeddings-ViT-Large-Patch14-336/version/3?tid=72f988bf-86f1-41af-91ab-2d7cd011db47)

Model Publishing Pipeline: [run](https://ml.azure.com/experiments/id/68938786-c866-4386-b50f-785a0c7e64db/runs/cool_cart_d9qhj72x3v?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rbhimani-rg/providers/Microsoft.MachineLearningServices/workspaces/rbhimani-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#)

### CODE REVIEW

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+tags:
+    Preview: ''
+    huggingface_model_id: openai/clip-vit-large-patch14-336
+    SharedComputeCapacityEnabled: ''
+    author: openai
+    license: mit
+    task: embeddings
+    inference_compute_allow_list:
+        [
+            Standard_DS2_v2,
```
Comment (rdondera-microsoft) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/spec.yaml:27]: Just making sure: did you test CLIP-L@336 on Ds2_v2?

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        "image_features": [0.92, -0.13, 0.02, ... , -0.13],
+        "text_features": [0.42, 0.13, -0.92, ... , -0.63]
+    },
+    {
+        "image_features": [-0.54, -0.83, 0.13, ... , -0.26],
+        "text_features": [-0.14, -0.93, 0.15, ... , 0.66]
+    }
+]
+```
+Note: returned embeddings have dimension 768 and are not normalized
```
Comment (rdondera-microsoft) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:109]: 768 -> 1024

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+[
+    {
+        "image_features": [-0.92, -0.13, 0.02, ... , 0.13],
+    },
+    {
+        "image_features": [0.54, -0.83, 0.13, ... , 0.26],
+    }
+]
+```
+Note: returned embeddings have dimension 768 and are not normalized
```
Comment (rdondera-microsoft) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:44]: 768 -> 1024

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+[
+    {
+        "text_features": [0.42, -0.13, -0.92, ... , 0.63],
+    },
+    {
+        "text_features": [-0.14, 0.93, -0.15, ... , 0.66],
+    }
+]
+```
+Note: returned embeddings have dimension 768 and are not normalized
```
Comment (rdondera-microsoft) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:75]: 768 -> 1024

**Review by rdondera-microsoft — DISMISSED** [review]
LGTM - minor comments.

**Review by yiyoulin — DISMISSED** [review]

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
+The `CLIP` model was developed by OpenAI researchers to learn about what contributes to robustness in computer vision tasks and to test the ability of models to generalize to arbitrary image classification tasks in a zero-shot manner. The model uses a ViT-L/14 Transformer architecture trained at 336x336 pixel resolution as an image encoder and uses a masked self-attention Transformer as a text encoder. The model was trained on publicly available image-caption data, which was gathered in a mostly non-interventionist manner. The model is intended as a research output for research communities, and the primary intended users of these models are AI researchers. The model has been evaluated on a wide range of benchmarks across a variety of computer vision datasets, but it currently struggles with respect to certain tasks such as fine-grained classification and counting objects. The model also poses issues with regards to fairness and bias, and the specific biases it exhibits can depend significantly on class design and the choices one makes for categories to include and exclude.
```
Comment (gauravrajguru) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:1]: please align description as this [doc](https://microsoftapc-my.sharepoint.com/:w:/r/personal/ayushmishra_microsoft_com/_layouts/15/doc2.aspx?sourcedoc=%7B2E13DB38-A1A9-4148-B1DC-AC14A76E1F72%7D&file=System%20Registry%20model%20card%20alignment.docx&nav=eyJjIjo0MTgzNzU3OX0&action=default&mobileredirect=true&share=IQE42xMuqaFIQbHcrBSnbh9yAamNOJxrsZ9DvamUluUGp40&wdOrigin=TEAMS-ELECTRON.p2p_ns.bim&wdExp=TEAMS-CONTROL&wdhostclicktime=1703143808964&web=1).

Also you can refer to this [PR] (https://github.com/Azure/azureml-assets/pull/2010)

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        Standard_NC4as_T4_v3, Standard_NC6s_v3, Standard_NC8as_T4_v3, Standard_NC12s_v3,
+        Standard_NC16as_T4_v3, Standard_NC24s_v3, Standard_NC64as_T4_v3, Standard_NC24ads_A100_v4,
+        Standard_NC48ads_A100_v4, Standard_NC96ads_A100_v4, Standard_ND96asr_v4, Standard_ND96amsr_A100_v4,
+        Standard_ND40rs_v2
+tags:
+    Preview: ''
+    huggingface_model_id: openai/clip-vit-large-patch14-336
+    SharedComputeCapacityEnabled: ''
+    author: openai
+    license: mit
```
Comment (gauravrajguru) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/spec.yaml:23]: can add `training_dataset` info if available?

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/spec.yaml (hunk 1 above)
Comment (bhimar) [reply assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/spec.yaml:27]: yes, I used DS2_v2 as the inference compute for the publishing pipeline

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md (hunk 2 above)
Comment (bhimar) [reply assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:109]: image
The embedding dimension is 768. I have also verified this by inspecting the embeddings returned from the endpoint.

**Review by yiyoulin — APPROVED** [review]

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+
+### Inference samples
+
+Inference type|Python sample (Notebook)|CLI with YAML
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-sdk-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-cli-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.sh</a>
+Batch|<a href="https://aka.ms/azureml-infer-batch-sdk-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-batch-cli-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.sh</a>
+
+### Sample inputs and outputs (for real-time inference)
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:43]:
```suggestion
# Sample input and output (for real-time inference)
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+# License
+The MIT License is a permissive free software license originating at the Massachusetts Institute of Technology (MIT). The license allows users to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the software, subject to the conditions that the copyright notice and permission notice appear in all copies or substantial portions of the software.
+
+
+### Inference samples
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:36]:
```suggestion
# Inference samples
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+Inference type|Python sample (Notebook)|CLI with YAML
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-sdk-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-cli-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.sh</a>
+Batch|<a href="https://aka.ms/azureml-infer-batch-sdk-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-batch-cli-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.sh</a>
+
+### Sample inputs and outputs (for real-time inference)
+
+#### Sample input for image embeddings
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:45]:
```suggestion
### Sample input for image embeddings
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+      "data":[
+         ["image1", ""],
+         ["image2", ""]
+      ]
+   }
+}
+```
+Note: "image1" and "image2" should be publicly accessible urls or strings in `base64` format
+
+#### Sample output
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:63]:
```suggestion
### Sample output
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md (hunk 9 above)
Comment (rjaincc) [reply assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:45]: We should perhaps add a section for text embeddings for completeness.

Comment (rjaincc) [reply assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:45]: NVM, its there.

Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:45]:
```suggestion
### Sample input for image embeddings
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+        "image_features": [-0.92, -0.13, 0.02, ... , 0.13],
+    },
+    {
+        "image_features": [0.54, -0.83, 0.13, ... , 0.26],
+    }
+]
+```
+Note: returned embeddings have dimension 768 and are not normalized
+
+#### Sample input for text embeddings
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:77]:
```suggestion
### Sample input for text embeddings
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+      "index":[0, 1],
+      "data":[
+         ["", "sample text 1"],
+         ["", "sample text 2"]
+      ]
+   }
+}
+```
+
+#### Sample output
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:94]:
```suggestion
### Sample output for text embeddings
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+        "text_features": [0.42, -0.13, -0.92, ... , 0.63],
+    },
+    {
+        "text_features": [-0.14, 0.93, -0.15, ... , 0.66],
+    }
+]
+```
+Note: returned embeddings have dimension 768 and are not normalized
+
+#### Sample input for image and text embeddings
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:108]:
```suggestion
### Sample input for image and text embeddings
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+      "data":[
+         ["image1", "sample text 1"],
+         ["image2", "sample text 2"]
+      ]
+   }
+}
+```
+Note: "image1" and "image2" should be publicly accessible urls or strings in `base64` format
+
+#### Sample output
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:126]:
```suggestion
#### Sample output for image and text embeddings
```

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+The MIT License is a permissive free software license originating at the Massachusetts Institute of Technology (MIT). The license allows users to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the software, subject to the conditions that the copyright notice and permission notice appear in all copies or substantial portions of the software.
+
+
+### Inference samples
+
+Inference type|Python sample (Notebook)|CLI with YAML
+|--|--|--|
+Real time|<a href="https://aka.ms/azureml-infer-sdk-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-cli-image-text-embeddings" target="_blank">image-text-embeddings-online-endpoint.sh</a>
+Batch|<a href="https://aka.ms/azureml-infer-batch-sdk-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.ipynb</a>|<a href="https://aka.ms/azureml-infer-batch-cli-image-text-embeddings" target="_blank">image-text-embeddings-batch-endpoint.sh</a>
+
```
Comment (rjaincc) [inline assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:42]: For Sample I/O, format that's followed else where:

```
# Sample input and output

### Sample input

### Sample output


#### Visualization of inference result for a sample image
```

**Review by rjaincc — APPROVED** [review]
Approving with some suggestions

File: assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md (hunk 7 above)
Comment (novaturient95) [reply assets/models/system/openai-clip-image-text-embeddings-vit-large-patch14-336/description.md:43]: Please remove: `(for real-time inference)` from the end as well

**Review by yiyoulin — APPROVED** [review]

**Pull request merged**